src/main/java/com/example/naitei19javaecommerce/service/Impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.example.naitei19javaecommerce.service.Impl;
+
+import com.example.naitei19javaecommerce.model.Product;
+import com.example.naitei19javaecommerce.repository.InvoiceDetailRepository;
+import com.example.naitei19javaecommerce.repository.ProductRepository;
+import com.example.naitei19javaecommerce.service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+
+    @Autowired
+    private InvoiceDetailRepository invoiceDetailRepository;
+
+    @Autowired
+    private ProductRepository productRepository;
+
+    @Override
+    public List<Product> getOutStandingProducts() {
+        List<Product> outStandingProducts = new ArrayList<>();
+        List<Long> productIds = invoiceDetailRepository.findOutStandingProductId();
+
+        if (productIds.size() > 5) {
+            productIds = productIds.subList(0, 5);
+        }
+
+        for (Long productId : productIds) {
+            Product product = productRepository.findOutStandingProduct(productId);
+            outStandingProducts.add(product);
+        }
```

Comment:
cực quá cụ, dùng where IN đi, tự nhiên đi query 5 lần
đam mê quá hen

Reply:
trời ơi em query in trong 1 câu sql nhưng mà nó không cho c ưi ,em kb sao nhưng mà nó k cho nên em mới phải query 2 lần đó huhu

Reply:
![image](https://github.com/awesome-academy/hn_naitei19-05-ecommerce/assets/95512496/de2b85ed-b324-444d-b3d0-fb3773bb1eea)
 ban đầu e làm như niii